httpmux/http_plugin_impl.go

```diff
@@ -2,41 +2,50 @@ package httpmux
 
 import (
 	"fmt"
+	"net/http"
+	"time"
+
 	"github.com/gorilla/mux"
 	"github.com/ligato/cn-infra/core"
-	"github.com/ligato/cn-infra/datasync"
 	"github.com/ligato/cn-infra/datasync/rpc/grpcsync"
-	"github.com/ligato/cn-infra/datasync/syncbase"
 	"github.com/ligato/cn-infra/logging"
 	"github.com/ligato/cn-infra/utils/safeclose"
 	"github.com/namsral/flag"
 	"github.com/unrolled/render"
-	"net/http"
-	"time"
 )
 
-// PluginID used in the Agent Core flavors
-const PluginID core.PluginName = "HTTP"
-
 const (
-	// DefaultHTTPPort is used during HTTP server startup unless different port was configured
+	// PluginID used in the Agent Core flavors
+	PluginID core.PluginName = "HTTP"
+	// DefaultHTTPPort is used as a default HTTP rest server startup unless different port was configured
 	DefaultHTTPPort = "9191"
+	// DefaultProbePort is used as a default HTTP probe server startup unless different port was configured
+	DefaultProbePort = "9190"
```

Comment:
Please move this flag to generic flavor (we do not want to uglify the httpmux based on statuscheck related requirements). Then inject the value to httpmux. There should be also logic in generic flavor that compares if probe port & http port is different. Keep the values same please by default. If the values are same then do not create new instance of httpmux but if it is different inject new instance.

Two more extra work items:
1. create dedicated structure for httpmux configuration with string endpoint istead of http port. This new structure will replace HTTPPort injectable field.
2.  move status-check to health/status-check. By doing this we have consistent diagram in root README.md with package hierarchy and we are ready to extract liveness & readiness plugin from status-check.